engine/src/main/java/org/terasology/engine/core/PathManager.java

```diff
@@ -355,7 +355,7 @@ private void updateDirs() throws IOException {
         Path homeModPath = homePath.resolve(MODULE_DIR);
         Path modCachePath = homePath.resolve(MODULE_CACHE_DIR);
 
-        if (homePath == installPath) {
+        if (homePath.equals(installPath)) {
```

Comment:
This is a cross-language confusion thing that I probably keep tripping over.

Note to self:

Java `==` compares only instance identity. More like C. Python `is` or JavaScript `===` (I think).

Java does _not_ support operator overloading at all. Defining `.equals` is not like defining Python `__eq__`.